### ISSUE TITLE
[a11y]: Downshift a11y helper text causes error

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#9841]
### Package

carbon-components-react

### Package version

latest

### Automated testing tool and ruleset

JAWS

### Description

Regarding the input…. the error message is coded as:
`<div role="alert" class="bx--form-requirement" id="comment_1.1.1-error-msg">External comment is required</div>`

"The default of an alert is aria-live=assertive. So, if you set the error to show when the input is blurred… you tab off of the input, the alert is announced, and the thing that you tabbed to is not announced at all (squashed by the assertive). I /think/ adding aria-live="polite" will ensure the message is announced, but won’t squash the announcement of the thing I’ve tabbed to. Unfortunately, I don’t have a great way of really testing this to confirm what it would do as polite."

### CodeSandbox example

https://codesandbox.io/s/github/carbon-design-system/carbon/tree/main/packages/react/examples/codesandbox

### Steps to reproduce

see above

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
